File: Core/Core.Utilities/TransactionHandler.vb
Imports JestersCreditUnion.DataTier.Utilities

Public Class TransactionHandler
    Inherits Settings
    Implements JestersCreditUnion.DataTier.Utilities.ITransactionHandler

    Private m_transactionHandler As Framework.ITransactionHandler

    Public Sub New(ByVal transactionHandler As Framework.ITransactionHandler)
        MyBase.New(transactionHandler)
        m_transactionHandler = transactionHandler
    End Sub

    Private Property DbConnection As IDbConnection Implements ITransactionHandler.Connection
        Get
            Return m_transactionHandler.DbConnection
        End Get
        Set(value As IDbConnection)
            m_transactionHandler.DbConnection = value
        End Set
    End Property

    Private Property DbTransaction As IDbTransaction Implements ITransactionHandler.Transaction
        Get
            Return m_transactionHandler.DbTransaction
        End Get
        Set(value As IDbTransaction)
            m_transactionHandler.DbTransaction = value
        End Set
    End Property
End Class

File: API/CommonCore/TransactionHandler.cs
using System.Data.Common;
using System.Threading.Tasks;

namespace JestersCreditUnion.CommonCore
{
    public class TransactionHandler : ITransactionHandler
    {
        private readonly ISettings _settings;

        public TransactionHandler(ISettings settings)
        {
            _settings = settings;
        }

        public DbConnection Connection { get; set; }
        public BrassLoon.DataClient.IDbTransaction Transaction { get; set; }

        public Task<string> GetConnectionString() => _settings.GetConnetionString();
    }
}
